src/conn_state.h

```diff
@@ -0,0 +1,50 @@
+/*
+* Copyright 2019 Xaptum, Inc.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+#pragma once
+
+#ifndef ENFTUN_CONNECTION_STATE_H
+#define ENFTUN_CONNECTION_STATE_H
+
+#include "netlink.h"
+
+#include <uv.h>
+
+struct enftun_conn_state;
+
+typedef void (*enftun_conn_state_reconnect)(struct enftun_conn_state* conn_state);
+
+struct enftun_conn_state
+{
+    uv_poll_t poll;
+    void* data;
```

Comment:
What is the `void* data` pointer for?

Reply:
It is used by the `trigger_reconnect` function to access the context. The naming came from the way the callback function was set up in `chain.h`. However, now that I include `context.h`, it could be of `enftun_context` type.

Reply:
I see, thanks.

No need to change it; `void *` is better.  We don't want to tie the `enftun_conn_state` functions (and callbacks) to any particular use case or caller.

Maybe add a comment like so:
```
void* data; // data for reconnect callback
```

Not having a similar comment in the chains is my fault.